frame/support/procedural/src/pallet/parse/hooks.rs

```diff
@@ -66,10 +68,16 @@ impl HooksDef {
 			return Err(syn::Error::new(item_trait.span(), msg));
 		}
 
+		let has_runtime_upgrade = item.items.iter().any(|i| match i {
+			syn::ImplItem::Method(method) => method.sig.ident == "on_runtime_upgrade",
```

Comment:
Happy to write a test for this, if someone can point out where, or find another way to avoid hardcoding the name here.